datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -99,19 +102,45 @@ pub trait Cursor: Ord {
 
     /// Advance the cursor, returning the previous row index
     fn advance(&mut self) -> usize;
+
+    /// Slice the cursor at a given row index, returning a new cursor
+    ///
+    /// Returns an error if the slice is out of bounds, or memory is insufficient
+    fn slice(&self, offset: usize, length: usize) -> Result<Self>
```

Comment:
Does this interface need to be fallible? It looks like the only error is out of bounds, which would imply a logic error at which point I think a panic is fine?